### PR TITLE
Ghost variable reference documentation

### PR DESCRIPTION
**Pull request by mdgeorge4153**
I added reference material about ghost variables, and updated the examples to use ghost variables instead of ghost functions.

### CODE REVIEW

File: docs/ref-manual/cvl/ghosts.md [hunk 1]
```diff
@@ -2,9 +2,117 @@
 Ghosts
 ======
 
+Ghosts are a way of defining global variables for use during verification.
```
Comment (nd-certora) [inline docs/ref-manual/cvl/ghosts.md:5]: not sure about global. how about "additional"  or "Auxiliary"

**Review by nd-certora — APPROVED** [review]
nice

File: docs/ref-manual/cvl/ghosts.md (hunk 1 above)
Comment (mdgeorge4153) [reply docs/ref-manual/cvl/ghosts.md]: Changed to additional

**Review by alexandernutz — COMMENTED** [review]
makes sense to me (modulo some small comments)

**Comment by alexandernutz** [comment]
d'oh .. an hour late ..

File: docs/ref-manual/cvl/ghosts.md [hunk 2]
```diff
@@ -2,9 +2,117 @@
...
+These variables are often used to communicate information between
+[rules](rules.md) and [hooks](hooks.md).
+
 ```
-ghost ::= "ghost" "(" [ cvl_type { "," cvl_type } ] ")"
-          ( ";" | "{" axiom { axiom } "}" )
+ghost ::= "ghost" type id               (";" | "{" axioms "}")
+        | "ghost" id "(" cvl_types ")"  (";" | "{" axioms "}")
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/ref-manual/cvl/ghosts.md:17]: I just noticed this and committed a fix in master.

File: docs/ref-manual/cvl/ghosts.md [hunk 3]
```diff
@@ -2,9 +2,117 @@
...
+This documentation is incomplete.  See [the old documentation](/docs/confluence/anatomy/ghostfunctions)
+for information about initial state axioms.
+```
+
+(ghost-axioms)=
+Ghost functions
+---------------
+
+CVL also has support for "ghost functions".  These serve a different purpose
+from ghost variables, although they can be used in similar ways.
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/ref-manual/cvl/ghosts.md:117]: yes =)

File: docs/user-guide/map/iterable.md [hunk 4]
```diff
@@ -142,48 +142,71 @@ invariant inMapIffInArray(uint x)
...
-invariant checkMapGhost(uint someKey) get(someKey) == _map(someKey)
+invariant checkMapGhost(uint someKey) get(someKey) == _map[someKey]
 ```
 
 Currently, the rule fails for all state-mutating functions, and even in the contract's initial state after constructor (rule `checkMapGhost_instate`):
 
 ![](ghost_fail.png)
 
-This is, in fact, unsurprising. There is nothing in the spec that links the value of the ghost to its Solidity counterpart. To make that link, we write _hooks_. Hooks allow us to instrument the verified code, that is, to wrap a bytecode operation with our own code, defined in the spec file.
+This is, unsurprising. There is nothing in the spec that links the
```
[the remark this answers is not in the document]
Comment (mdgeorge4153) [reply docs/user-guide/map/iterable.md:176]: fixed in master, thanks

**Pull request merged**